R/getvectorofcolors.R

```diff
@@ -0,0 +1,137 @@
+#'  gradient when \code{palette} is set to \code{"Custom gradient"}.
+#' @param palette.custom.palette A vector or comma separated list of colors 
+#'  which will be recycled to the desired length.
+#' @param color.values An optional numeric vector or matrix which can
+#'  be used with gradual palettes (either a custom gradient or one of
+#'  the sequential color palettes). The names of the vector or matrix
+#'  should be the same as the names of \code{input.data}, and
+#'  the colorramp is constructed across the range of the values.
+#' @param small.multiples Logical indicating whether the \code{color.values}
+#'  are being applied to for small multiples (a matrix of colors is 
+#'  created instead of a vector
+#' @param type Describes the type of data which the color vector will be applied to.
+#'  One of "Series" or "Pie subslice".
+#' @importFrom grDevices colorRamp
+#' @export
+GetVectorOfColors <- function (template,
+                               input.data, 
+                               filter,
+                               chart.type, 
+                               scatter.colors.column = FALSE, 
+                               multi.color.series = FALSE,
+                               palette = NULL,
+                               palette.custom.color = NULL, 
+                               palette.custom.gradient.start = NULL,
+                               palette.custom.gradient.end = NULL,
+                               palette.custom.palette = NULL,
+                               color.values = NULL,
+                               small.multiples = FALSE,
+                               type = "Series")
+{
+    if (is.null(palette))
+        return (NULL)
+    if (palette == "Group colors")
+        return (NULL)
+    if (type == "Pie subslice" && chart.type != "Pie")
+        return (NULL)
+    if (type == "Pie subslice" && NCOL(input.data) == 1)
+        return (NULL)
+
+    # This step converts "Named colors" into a vector
+    # But leaves everything else unchanged
+    if (is.null(template$brand.colors) && !is.null(names(template$colors)))
+        template$brand.colors <- template$colors
+    tmp.palette <- GetPalette(palette, template)
+
+    # Get vector of colors
+    index <- if (type == "Pie subslice") 2 else 1
+    num.colors <- GetNumColors(input.data, chart.type, scatter.colors.column, multi.color.series)[[index]]
+    unordered.colors <- ChartColors(num.colors, given.colors = tmp.palette, 
+        custom.color = palette.custom.color,
+        custom.gradient.start = palette.custom.gradient.start, 
+        custom.gradient.end = palette.custom.gradient.end,
+        custom.palette = palette.custom.palette,
+        silent = chart.type %in% c("Pie", "Donut"),
+        silent.single.color = multi.color.series || 
+            chart.type %in% c("Bar Pictograph", "Time Series", "Pyramid"))
+
+    series.names <- GetBrandsFromData(input.data, filter, chart.type, 
```

Comment:
I would prefer not to add a new function name, because that makes it even more complicated. Firstly in terms of adding a extra function call layer, and secondly when you do a search on a function name and realise you need to look for another term.
I have added some more info to the description to clarify that "Brand names" is actually a shorthand for names associated with the colors in https://github.com/Displayr/flipChartBasics/commit/94c5264eec55351515c6a15d615818ee75e27f90#diff-d96f71e188b5e83548bd6590e6872efedc2d61eb5709394d5c20a257aa3bbfa7L87

Also note that `GetBrandsFromData` is used inside `GetVectorOfColors`, but hopefully, it shouldn't need to be called directly inside the Standard R Code.